### PR TITLE
Vehicle update: Vehicle2

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Complete. Doesn't desync on lets play save but since it has a good bit of rack rail and manual control code could do with testing.

### CODE REVIEW

**Comment by LeftofZen** [comment]
I'm happy to test this after merge conflicts are fixed

File: src/OpenLoco/Vehicles/Vehicle2.cpp [hunk 1]
```diff
@@ -1,16 +1,353 @@
...
 namespace OpenLoco::Vehicles
 {
+    static loco_global<uint32_t, 0x01136114> vehicleUpdate_var_1136114;
+    static loco_global<int32_t, 0x0113612C> vehicleUpdate_var_113612C; // Speed
+    static loco_global<int32_t, 0x01136130> vehicleUpdate_var_1136130; // Speed
+    static loco_global<Speed32, 0x01136134> vehicleUpdate_var_1136134; // Speed
+    static loco_global<VehicleHead*, 0x01136118> vehicleUpdate_head;
+
+    // values are pre *256 for maths
+    static constexpr std::array<int32_t, 13> _500170{
```
Comment (LeftofZen) [inline src/OpenLoco/Vehicles/Vehicle2.cpp:21]: what are these values for?

Comment (duncanspumpkin) [reply src/OpenLoco/Vehicles/Vehicle2.cpp:21]: I'm not entirely sure but think they are something to do with amount of negative force due to the angle of the track and gravity.
So its like `256 * 256 * SIN(angle)`

Comment (LeftofZen) [reply src/OpenLoco/Vehicles/Vehicle2.cpp:21]: Interesting. There is a strange pattern as well:
- ignore first zero
- 4 values in position 1->5
- next four values are the negative of the first four
- next 2 values are negatives of each other
- last 2 values also negatives of each other

Comment (LeftofZen) [reply src/OpenLoco/Vehicles/Vehicle2.cpp:21]: ah so it does, that's useful to know, maybe we should add a comment mentioning these array values follow the sprite pitches?

**Review by LeftofZen — DISMISSED** [review]

**Comment by LeftofZen** [comment]
I playtested this and manual driving is rather hilarious - braking has the same effect as accelerating, so your train just goes full-stream ahead and crashes into whatever is in front of it.

**Comment by LeftofZen** [comment]
Everything seems to work as expected now.

**Review by LeftofZen — APPROVED** [review]
Everything seems to work as expected now.

**Pull request merged**